### PR TITLE
add support for framework v1.1.1-1.10.4

### LINKED ISSUE #88: Support 1.1.0-1.10.3 or newer
Due to the [1.0.1-1.10.3 breaking changes](https://docs.mesosphere.com/services/kubernetes/1.1.0-1.10.3/release-notes/#breaking-changes), there needs to be some work on this quickstart to automate exposing the Kubernetes API.

### PR DESCRIPTION
**Pull request by rimusz**
includes kubernetes API access via marathon-lb

Fixes #88

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,31 +126,40 @@ launch-dcos: check-terraform
...
+.PHONY: plan
+plan: plan-dcos
+
+.PHONY: deploy
+deploy: check-cli launch-dcos setup-cli install
+
+.PHONY: setup-cli
+setup-cli: check-dcos
+	@echo "Sleep for 15 seconds ..."
+	@sleep 15
```
Comment (bmcstdio) [inline Makefile]: @rimusz why do we need to sleep?

File: Makefile [hunk 2]
```diff
@@ -124,31 +126,40 @@ launch-dcos: check-terraform
...
+	watch dcos kubernetes plan show deploy
 
-.PHONY: deploy
-deploy: check-cli launch-dcos setup-cli install
+.PHONY: kubeconfig
+kubeconfig:
+	$(call get_public_agent_ip)
+	$(DCOS_CMD) kubernetes kubeconfig --apiserver-url https://$(PUBLIC_AGENT_IP):6443 --insecure-skip-tls-verify
+	@echo "Sleep for 25 seconds ..."
+	@sleep 25
```
Comment (bmcstdio) [inline Makefile]: @rimusz why do we need to sleep?

File: docs/aws.md [hunk 3]
```diff
@@ -73,6 +73,28 @@ For more advanced scenarios, please check the [terraform-dcos documentation for
 
 ### Kubernetes configuration
 
+#### RBAC
+
+**NOTE:** By default, it will provision a Kubernetes cluster without [RBAC](https://docs.mesosphere.com/services/kubernetes/1.1.1-1.10.4/authn-and-authz/#rbac) support.
```
Comment (bmcstdio) [inline docs/aws.md]: @rimusz perhaps `s/it/dcos-kubernetes-quickstart`.

File: docs/aws.md [hunk 4]
```diff
@@ -89,6 +111,8 @@ To deploy a **highly-available** cluster with three (3) private and one (1) publ
 }
 ```
 
+**NOTE:** The **highly-available** mode for a cluster must be chosen when installing the package. Changing the highly-available mode after installing the package is not supported.
```
Comment (bmcstdio) [inline docs/aws.md]: @rimusz please see the comment above about `kubernetes.high_availability`.

File: docs/azure.md [hunk 5]
```diff
@@ -68,6 +68,27 @@ For more advanced scenarios, please check the [terraform-dcos documentation for
 
 ### Kubernetes configuration
 
+#### RBAC
+
+**NOTE:** By default, it will provision a Kubernetes cluster without [RBAC](https://docs.mesosphere.com/services/kubernetes/1.1.1-1.10.4/authn-and-authz/#rbac) support.
```
Comment (bmcstdio) [inline docs/azure.md]: @rimusz perhaps `s/it/dcos-kubernetes-quickstart/`.

File: docs/azure.md [hunk 6]
```diff
@@ -83,6 +104,8 @@ To deploy a **highly-available** cluster with three (3) private and one (1) publ
 }
 ```
 
+**NOTE:** The **highly-available** mode for a cluster must be chosen when installing the package. Changing the highly-available mode after installing the package is not supported.
```
Comment (bmcstdio) [inline docs/azure.md]: @rimusz please see the comment above about `kubernetes.high_availability`.

File: resources/kubeapi-proxy.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+  "id": "/kubeapi-proxy",
```
Comment (bmcstdio) [inline resources/kubeapi-proxy.json]: @rimusz I got `kubeapi-proxy` as well as `marathon-lb`. Could we stick to a single one or have that as as an option of the Makefile target?

File: README.md [hunk 8]
```diff
@@ -6,11 +6,11 @@ Kubernetes is now available as a DC/OS package to quickly, and reliably run Kube
 
 ## Known limitations
 
-Before proceeding, please check the [current package limitations](https://docs.mesosphere.com/service-docs/kubernetes/1.0.3-1.9.7/limitations/).
+Before proceeding, please check the [current package limitations](https://docs.mesosphere.com/service-docs/kubernetes/1.1.1-1.10.4/limitations/).
 
 ## Pre-Requisites
 
-First, make sure your cluster fulfils the [Kubernetes package default requirements](https://docs.mesosphere.com/service-docs/kubernetes/1.0.3-1.9.7/install/#prerequisites/).
+First, make sure your cluster fulfils the [Kubernetes package default requirements](https://docs.mesosphere.com/service-docs/kubernetes/1.1.1-1.10.4/install/#prerequisites/).
```
[the remark this answers is not in the document]
Comment (rimusz) [reply README.md]: one has enough cloud resources

File: Makefile (hunk 1 above)
Comment (rimusz) [reply Makefile]: to wait for master be ready, specially it happens on was

File: Makefile (hunk 2 above)
Comment (rimusz) [reply Makefile]: same here, better put some delay, as otherwise kubeapi is not available right after the above command

File: Makefile (hunk 1 above)
Comment (pires) [reply Makefile]: Is 15 seconds enough?

File: Makefile (hunk 2 above)
Comment (pires) [reply Makefile]: Can we poll until the Kubernetes API is up instead?

File: docs/aws.md (hunk 3 above)
Comment (pires) [reply docs/aws.md]: I suggest replacing _it will provision_ with _this quickstart provisions_.

File: docs/azure.md (hunk 5 above)
Comment (pires) [reply docs/azure.md]: Why are we repeating text in all docs?

File: resources/kubeapi-proxy.json (hunk 7 above)
Comment (rimusz) [reply resources/kubeapi-proxy.json]: we need both, as `kube-proxy` is just a dummy marathon app with traffic forward to k8s API

Comment (pires) [reply resources/kubeapi-proxy.json]: Makes sense.

**Pull request merged**